### PR TITLE
Ui redesign

### PR DESCRIPTION
**Pull request by utsavgarg**
#46  #14 #16

### CODE REVIEW

File: ide/static/css/style.css [hunk 1]
```diff
@@ -73,6 +73,21 @@ html,body,#app,.app{
...
+.customTooltip {
+  padding: 10px !important;
+  width: 270px;
+  color: white !important;
+  background-color: rgba(0,0,0,0.9) !important;
+  border-radius: 20px;
+}
+
+.customTooltip br {
+display: none;
```
Comment (virajprabhu) [inline ide/static/css/style.css]: indent

File: ide/static/css/style.css [hunk 2]
```diff
@@ -73,6 +73,21 @@ html,body,#app,.app{
...
 }
 
+.rhTooltip{
+  padding: 0px;
+}
+
+.customTooltip {
+  padding: 10px !important;
+  width: 270px;
+  color: white !important;
```
Comment (virajprabhu) [inline ide/static/css/style.css]: Is the !important really required? Makes debugging harder in general.

File: ide/static/js/content.js [hunk 3]
```diff
@@ -251,8 +269,8 @@ class Content extends React.Component {
     Object.keys(positions).forEach(layerId => {
       const layer = net[layerId];
       layer.state = {
-        top: `${250 + 50 * positions[layerId][1]}px`,
-        left: `${20 + 180 * positions[layerId][0]}px`,
+        top: `${height + 65 * positions[layerId][1]}px`,
```
Comment (virajprabhu) [inline ide/static/js/content.js]: We can ignore this issue (centering fails on some large screens) for now, I guess. Please add a comment here documenting it.

File: ide/static/js/panZoom.js [hunk 4]
```diff
@@ -86,38 +82,20 @@ export default function() {
...
-        return p[1]; 
-      } } 
-      return(false); 
-    }
-
+	var params = window.location.search.substring(1).split("&");  
+	for (var i = 0; i < params.length; i++) {
+		var p = params[i].split("="); 
+		if (p[0] == id) { 
+			return p[1]; } }
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: Fix indenting of this entire block.

File: ide/static/js/netLayout_vertical.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+let processed = {};
+
+Object.keys(net).forEach(layerId => {
+  processed[layerId] = false;
+});
+
+function isProcessPossible(layerId){
+  let inputs = net[layerId].connection.input;
+  let i = 0;
+  for(i = 0; i < inputs.length; i++){
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: space after for

File: ide/static/js/netLayout_vertical.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    if (processed[inputs[i]] === false) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// allocatePosition finds the closest position available to preferred position
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: Please add a more descriptive comment for how this works

File: ide/static/js/netLayout_vertical.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  layerId = stack[i];
+  stack.splice(i, 1);
+  parentId = parentMap[layerId];
+  inputLength = net[layerId].connection.input.length;
+  if (parentId != null){
+    outputLength = net[parentId].connection.output.length;
+  }
+  if (parentId === null) {
+    position[layerId] = [0,0];
+  } else if(inputLength === 1 && outputLength === 1){
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: space after else if

File: ide/static/js/netLayout_vertical.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+// finding the input layers to start DFS
+Object.keys(net).forEach(layerId => {
+  if (net[layerId].info.type === 'Data' || net[layerId].info.type === 'Input' || net[layerId].info.type === 'HDF5Data') {
+    stack.push(layerId);
+    parentMap[layerId] = null;
+  }
+});
+
+// custom DFS
```
Comment (virajprabhu) [inline ide/static/js/netLayout_vertical.js]: Please add a descriptive comment for the "custom DFS"

File: ide/static/js/panZoom.js [hunk 9]
```diff
@@ -68,14 +63,15 @@ export default function() {
       canvas.y += e.pageY - previousMousePosition.y;
       canvas.updateContainerPosition();
       previousMousePosition = { x: e.pageX, y: e.pageY };
-      //instance.repaintEverything();
     }
   };
 
+  panZoom.onT
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: ?

File: ide/static/js/panZoom.js [hunk 10]
```diff
@@ -68,14 +63,15 @@ export default function() {
...
+
   panZoom.ondblclick = function(e) {
     e.preventDefault();
-    onZoom((e.ctrlKey || e.metaKey) ? current.zoom * 1.7 * 1.7 : current.zoom / 1.7 / 1.7,
-     e.clientX - panZoom.offsetLeft, e.clientY - panZoom.offsetTop);
+    onZoom((e.ctrlKey || e.metaKey) ? current.zoom * 1.2 * 1.2 : 
+	current.zoom / 1.2 / 1.2, e.clientX - panZoom.offsetLeft, e.clientY - panZoom.offsetTop);
```
Comment (virajprabhu) [inline ide/static/js/panZoom.js]: Currently, double clicking zooms in and there is no way to zoom out. This should not be the case.

File: ide/static/js/tooltip.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import TooltipData from './tooltipData';
+import data from './data';
+import ReactTooltip from 'react-tooltip'
+
+class Tooltip extends React.Component {
```
Comment (virajprabhu) [inline ide/static/js/tooltip.js]: The tooltip implementation is neat! Nicely done.

File: ide/static/js/panZoom.js (hunk 10 above)
Comment (utsavgarg) [inline ide/static/js/panZoom.js]: No, Ctrl + double clicking zooms out. Also there is an option to use '[' to zoom in and ']' to zoom out.

Comment (virajprabhu) [inline ide/static/js/panZoom.js]: – If I zoom in with a double click, I'm unable to pinch out to zoom out.
– To confirm, with a mouse, are double-clicking/using '[' or ']' the only zooming hotkeys?

Comment (utsavgarg) [inline ide/static/js/panZoom.js]: If you zoom in with a double click, are you able to zoom out with CTRL + double click ?

**Pull request merged**